src/main/java/org/gridsuite/gateway/GatewayConfig.java

```diff
@@ -21,225 +18,88 @@
 @Configuration
 @PropertySource(value = {"classpath:allowed-issuers.yml"})
 @PropertySource(value = {"file:/config/allowed-issuers.yml"}, ignoreResourceNotFound = true)
-@EnableConfigurationProperties(UriConfiguration.class)
 public class GatewayConfig {
+
+    static final String HEADER_USER_ID = "userId";
+
     @Bean
-    public RouteLocator myRoutes(RouteLocatorBuilder builder, UriConfiguration uriConfiguration) {
+    public RouteLocator myRoutes(RouteLocatorBuilder builder, ServicesURIsConfig servicesURIsConfig) {
         return builder.routes()
             .route(p -> p
-                    .path("/study/**")
-                    .filters(f -> f.rewritePath("/study/(.*)", "/$1"))
-                    .uri(uriConfiguration.getStudyServerBaseUri())
+                .path("/study/**")
```

Comment:
still sad that path("/" + "<PATTERN>" + "/**").uri( "/" + "<PATTERN>").uri(service.get...)

Reply:
Done